api/test/wfl/tools/workloads.clj

```diff
@@ -259,13 +259,15 @@
         (when (> elapsed timeout)
           (throw (TimeoutException.
                   (format "Timed out waiting for workload %s" uuid))))
-        (if (or (:finished workload) (every? finished? (endpoints/get-workflows wl)))
-          (done! wl)
-          (do
-            (log/infof "Waiting for workload %s to complete" uuid)
-            (util/sleep-seconds interval)
-            (recur (+ elapsed interval)
-                   (endpoints/get-workload-status uuid))))))))
+        (let [workflows (endpoints/get-workflows wl)]
+          (if (or (:finished wl)
```

Comment:
This previously checked `(:finished workload)` but `workload` statically reflected the state of the workload at creation.

Reply:
+1